impl/maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultPluginDependenciesResolver.java

```diff
@@ -208,12 +221,29 @@ private DependencyResult resolveInternal(
             request.setRequestContext(REPOSITORY_CONTEXT);
             request.setRepositories(repositories);
             request.setRoot(new org.eclipse.aether.graph.Dependency(pluginArtifact, null));
+            Map<String, org.eclipse.aether.graph.Dependency> core = getCoreExportsAsDependencies(session);
+            request.setManagedDependencies(core.values().stream().toList());
             for (Dependency dependency : plugin.getDependencies()) {
                 org.eclipse.aether.graph.Dependency pluginDep =
                         RepositoryUtils.toDependency(dependency, session.getArtifactTypeRegistry());
                 if (!DependencyScope.SYSTEM.is(pluginDep.getScope())) {
                     pluginDep = pluginDep.setScope(DependencyScope.RUNTIME.id());
                 }
+                org.eclipse.aether.graph.Dependency managedDep =
+                        core.get(pluginDep.getArtifact().getGroupId() + ":"
+                                + pluginDep.getArtifact().getArtifactId());
+                if (managedDep != null) {
+                    // align version if needed
+                    if (!Objects.equals(
+                            pluginDep.getArtifact().getVersion(),
+                            managedDep.getArtifact().getVersion())) {
+                        pluginDep = pluginDep.setArtifact(pluginDep
+                                .getArtifact()
+                                .setVersion(managedDep.getArtifact().getVersion()));
+                    }
+                    // align scope
+                    pluginDep = pluginDep.setScope(managedDep.getScope());
```

Comment:
Should we flag them as `system` scope ?

Reply:
... to avoid downloading the jars ?

Reply:
Yes, but there is more to this PR, it breaks https://issues.apache.org/jira/browse/MNG-5783 We need to implement the proposed fix in refd issue (real classpath vs `${plugin.artifacts}`).

Reply:
Former contains plugin stuff + maven core (imported), while latter should contain all